pkg/router/router.go

```diff
@@ -98,7 +98,7 @@ func serve(ctx context.Context, logger *zap.Logger, port int,
 }
 
 // Start starts a router
-func Start(ctx context.Context, clientGen crd.ClientGeneratorInterface, logger *zap.Logger, port int, executorURL string) error {
+func Start(ctx context.Context, clientGen crd.ClientGeneratorInterface, logger *zap.Logger, port int, executor eclient.ClientInterface) error {
```

Comment:
Why did we move `executor eclient.ClientInterface` to the parameter?

Reply:
So that we can unit test router by mocking the executor calls.
This allows us to write tests with router and envtests and skipping rest of fission services.